### DISCUSSION TITLE
Fetching PostgreSQL tables without a PK

### DISCUSSION THREAD

**Discussion by IvanDrag0** [Budibase/budibase#11985] [Help] [answered]
It seems that Budibase cannot fetch tables that do not have a primary key. Is that by design? If that's the case, is it possible to force Budibase to fetch those tables without adding a PK.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @IvanDrag0,

To quickly answer your question, no its not forcible. You need to supply a table with a primary key in order to fetch/sync tables into Budibase.

Heres a [discussion](https://github.com/Budibase/budibase/discussions/11836#discussioncomment-7070342) were this was explained. To add additional context to this. Budibase needs primary keys, as they need a something unique to do CRUD related functions. Without primary keys you'd wouldn't be able to do the CRUD operations as their be no RowID to attach to the majority of the forms/actions. Relationships also wouldn't be useable as these depend on primary/foreign keys. Most of the automations require a primary key for updating rows etc as well. 

There is also some likely to be some backend related reason that primary keys are required.

I hope this additional context helps!

**Reply by IvanDrag0** [reply]
@ConorWebb96 Thank you for the explanation!
